### PR TITLE
Address post-release checklist items for 5.1

### PR DESCRIPTION
**Pull request by webvictim**
Draft PR pending the addition of a `quay.io/repository/gravitational/teleport:5` tag (as described in https://github.com/gravitational/teleport/issues/5216#issuecomment-754851672)

Depends on #5310

### CODE REVIEW

**Review by awly — APPROVED** [review]

**Comment by webvictim** [comment]
@klizhentas @benarent Needs a CODEOWNER approval for the docs changes please.

**Review by r0mant — APPROVED** [review]

**Review by benarent — APPROVED** [review]

**Pull request merged**
